templates/manage_certs.py

```diff
@@ -32,7 +33,7 @@ def get_all_domains(config):
             line = line.strip()
             if line and not line.startswith("#"):
                 domain, backend = line.split(None, 1)
-                domains[domain] = backend
+                domains[domain.lower()] = backend
```

Comment:
@antoviaque Most of the code can only be meaningfully unit tested by implementing a fake for the letsencrypt client, for haproxy, for the haproxy-config-watcher and the file system, due to the high level of integration of this code.  For this reason, I implemented integration tests with good test coverage instead, which also test all the other components of this role.  I think that this provides much more value for far less work.

I now added unit tests for a few functions, inlcuding this one, where we might get some limited value from such tests.